packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -110,24 +110,54 @@ export class MediaPlaceholder extends Component {
 
 	onFilesUpload( files ) {
 		const {
-			addToGallery,
 			allowedTypes,
 			mediaUpload,
 			multiple,
 			onError,
 			onSelect,
-			value = [],
 		} = this.props;
 		let setMedia;
 		if ( multiple ) {
-			if ( addToGallery ) {
```

Comment:
The add to gallery flag indicates that the media upload should be added to the existing media items, while if the flag is false the media uploaded should totally replace the existing media.
It seems we are removing the mechanism to totally replace existing media, should we keep the `if ( addToGallery ) {` condition and move our code inside the if?

Reply:
Since we are removing this condition, we are always assuming `addToGallery` is true here.  So we are always adding to the gallery, and never ‘replacing’.  If we look at how `addToGallery` prop is defined in `gallery/edit.js`, it is true whenever there are images in the gallery that contain IDs.  

This is problematic, consider the flow where we start with an empty gallery.  We add image1, `addToGallery` will be `false`.  While image1 is still uploading, we drag and drop image2 to start uploading.  Since image1 is not yet uploaded, there are no images with IDs in the gallery, and image2 starts its upload with `addToGallery` as false as well.  That is even though there is already an image in the gallery uploading (and our expectation for image2 is to ADD it to the gallery as opposed to replacing content), the second upload starts with the assumption that we are replacing content instead of adding.

Moreover, `addToGallery` in this implementation seems unnecessary in that its goal is to determine whether to ‘replace’ or ‘add’ the content, while the only time that it signifies replacing content is when the gallery is empty.  Adding to the gallery makes just as much sense as replacing in this circumstance.

I am a bit uncertain of the last place it is being used where it is passed into the `MediaUpload` component from the `MediaPlaceholder`.  It seems it wasn’t necessary to make any changes here to fix the behavioral issues when adding separate uploads to the empty gallery.  This uncertainty is one reason I removed that condition from the setMedia function instead of altering its definition in the gallery component to always be true (maybe it would make sense to do this instead?).

Reply:
>  instead of altering its definition in the gallery component to always be true (maybe it would make sense to do this instead?).

Actually, the more I think about it the more this makes sense.  So I added the check for `addToGallery` back into the `media-placeholder` component, and have edited the value we pass into this prop in `gallery/edit.js` to always be `true` for the reasons noted above.

Reply:
This solution seems good 👍